presto-hive/src/main/java/com/facebook/presto/hive/HivePageSourceProvider.java

```diff
@@ -422,6 +422,10 @@ protected static CacheQuota generateCacheQuota(HiveSplit hiveSplit)
             }
         }
 
+        if (!hiveColumns.isEmpty() && hiveColumns.get(0).getColumnType() == AGGREGATED) {
```

Comment:
get confuse here. if hiveColumns only have one AGGREGATED column, is it valid?

Reply:
Either all columns are aggregate or none is. So we just check the first

Reply:
get it. shall we use allMatch to make sure all are AGGREGATED?
I think we also need to update the message

Reply:
Converted to allMatch. What is your suggestion for error message? It reads ok to me